plugins/Actions/Actions/ActionSiteSearch.php

```diff
@@ -132,8 +132,20 @@ protected function detectSiteSearchFromUrl($website, $parsedUrl)
         $keywordParameters = isset($website['sitesearch_keyword_parameters'])
             ? $website['sitesearch_keyword_parameters']
             : array();
-        $queryString = (!empty($parsedUrl['query']) ? $parsedUrl['query'] : '') . (!empty($parsedUrl['fragment']) ? $separator . $parsedUrl['fragment'] : '');
-        $parametersRaw = UrlHelper::getArrayFromQueryString($queryString);
+        $queryString = !empty($parsedUrl['query']) ? $parsedUrl['query'] : '';
+        $fragment = !empty($parsedUrl['fragment']) ? $parsedUrl['fragment'] : '';
+
+        // check if fragment contains a separate query (beginning with ?) otherwise assume complete fragment as query
+        if ($fragment && strpos($fragment, '?') !== false &&
+            (strpos($fragment, '&') === false || strpos($fragment, '?') < strpos($fragment, '&'))) {
```

Comment:
I think the second condition is to differentiate between a fragment that is a query string and a fragment w/ `&`/`?` that isn't a query string, correct? I haven't tried it, but would it be easier to call `parse_url()` on the fragment and check for the `'query'` property?